### ISSUE TITLE
关于训练数据的问题 Decompressed Data Too Large

### ISSUE THREAD

**Issue by SailVR** [liujiaheng/compression#30]
作者，你好！你的代码对我很有帮助，但是你的训练数据实在太大，国内用户下载google drive太慢。所以我把你的dataset里面注释那行transforms.RandomResizedCrop(self.image_size)加上了，使用了256的裁剪。但是我训练数据使用121326张图片，大概迭代1000次（还在第0个epoch），就报pillow太大的错误了ValueError: Decompressed Data Too Large。但是将图片训练集改到856张的训练集后，就没有出现问题。

**Comment by liujiaheng** [comment]
那我不大清楚啊，我后续把数据也上传到百度云盘吧

**Comment by SailVR** [comment]
谢谢了

**Comment by SailVR** [comment]
目前我已经解决了这个问题，解决方案有俩种：

**Comment by SailVR** [comment]
1 pillow版本过高，降下版本
2 在定义的dataset里面添加try except模块，如果try不成功，就在except里面返回None
    def __getitem__(self, item):
        try:
            image_ori = self.image_path[item]
            image = Image.open(image_ori).convert('RGB')
            transform = transforms.Compose([
                transforms.RandomResizedCrop(self.image_size),
                transforms.RandomHorizontalFlip(),
                transforms.RandomVerticalFlip(),
                transforms.ToTensor(),
                # transforms.Normalize((5, 0.5, 0.5), (0.5, 0.5, 0.5))
            ])
            return transform(image)
        except:
            return None
然后自定义个函数exception_collate_fn
def exception_collate_fn(batch):
    batch = list(filter(lambda x: x is not None, batch))
    return torch.utils.data.dataloader.default_collate(batch)
在train.py的DataLoader里面参数添加collate_fn=exception_collate_fn
    train_loader = DataLoader(dataset=train_dataset,
                              batch_size=batch_size,
                              shuffle=True,
                              collate_fn=exception_collate_fn,
                              pin_memory=True,
                              num_workers=4)

**Comment by SailVR** [comment]
目前我把你的复现代码在我的数据集上训练了，迭代了一百五十万次，训练的结果与作者的原论文保持一致，非常感谢作者的复现。**希望作者尽快把EDIC的代码开源出来。**
